### ISSUE TITLE
socket won't reconnect after clicking icon.

### ISSUE THREAD

**Issue by sweeneyapps** [rainforestapp/tester-chrome-extension#706]
for some reason, socket get disconnected and couldn't reconnect after clicking the icon.

I will do the pull request for this fix.

**Comment by shosti** [comment]
@sweeneyapps did you get a chance to see if the socket was in "disconnected" or "left" state?

**Comment by sweeneyapps** [comment]
@shosti it was in "unconnected" state.

**Comment by sweeneyapps** [comment]
sent screenshot via email.

**Comment by sweeneyapps** [comment]
thanks @shosti ..    

https://github.com/rainforestapp/tester-chrome-extension/pull/712 fixes this.

**Comment by shosti** [comment]
I rolled back the "fix" because it caused massive problems, and I remember now why it was the way it was--the socket library handles re-connection automatically (in a sane way) so it should never require manual intervention. Do you recall how long the plugin was disconnected?

**Comment by sweeneyapps** [comment]
I noticed it 2 days ago whenever I let macbook go to sleep.  Then I click on icon to activate it and it stay grey.  I went ahead debug everything and noticed that it doesn't affect the socket reconnection anywhere in the code only if the state is at "left".   That's why I opened up the issue and try to fix it then realized that I went little off.   It's not behaving as it should. 

Anyway, then the plugin stop working like last night around 6pm our time because of 503 error.   I believe the socket gone mad. :)
